sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosBulkItemResponse.java

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.JsonSerializable;
+import com.azure.cosmos.implementation.batch.BatchExecUtils;
+import com.azure.cosmos.util.Beta;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import reactor.core.publisher.Flux;
+
+import java.time.Duration;
+import java.util.Map;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Response of a {@link CosmosItemOperation} request when processed using Bulk by calling
+ * {@link CosmosAsyncContainer#processBulkOperations(Flux, BulkProcessingOptions)}.
+ *
+ */
+@Beta(Beta.SinceVersion.V4_9_0)
+public final class CosmosBulkItemResponse {
+
+    private final String eTag;
+    private final double requestCharge;
+    private final ObjectNode resourceObject;
+    private final int statusCode;
+    private final Duration retryAfter;
+    private final int subStatusCode;
+    private final Map<String, String> responseHeaders;
+    private final CosmosDiagnostics cosmosDiagnostics;
+
+    /**
+     * Initializes a new instance of the {@link CosmosBulkItemResponse} class.
+     */
+    CosmosBulkItemResponse(String eTag,
+                           double requestCharge,
+                           ObjectNode resourceObject,
+                           int statusCode,
+                           Duration retryAfter,
+                           int subStatusCode,
+                           Map<String, String> responseHeaders,
+                           CosmosDiagnostics cosmosDiagnostics) {
+
+        checkNotNull(responseHeaders, "expected non-null responseHeaders");
+
+        this.eTag = eTag;
+        this.requestCharge = requestCharge;
+        this.resourceObject = resourceObject;
+        this.statusCode = statusCode;
+        this.retryAfter = retryAfter;
+        this.subStatusCode = subStatusCode;
+        this.responseHeaders = responseHeaders;
+        this.cosmosDiagnostics = cosmosDiagnostics;
+    }
+
+    /**
+     * Gets the activity ID that identifies the server request made to execute this operation.
+     *
+     * @return the activity ID that identifies the server request made to execute this operation.
+     */
+    public String getActivityId() {
+        return BatchExecUtils.getActivityId(this.responseHeaders);
+    }
+
+    /**
+     * Gets the entity tag associated with the current item.
+     *
+     * ETags are used for concurrency checking when updating resources.
+     *
+     * @return Entity tag associated with the current item.
+     */
+    public String getETag() {
```

Comment:
how will someone use an ETAG in the bulk operation context? I don't think it can be used...

Reply:
Neve mind... the CosmosBulkItemResponse is per each operation in the bulk, right?

Reply:
Yes. And like normal operations, etag will yield NotModified or condition failed